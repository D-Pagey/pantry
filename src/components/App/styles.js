import styled, { createGlobalStyle } from 'styled-components/macro';
import { normalize } from 'polished';
import { NavLink } from 'react-router-dom';

export const GlobalStyle = createGlobalStyle`

  ${normalize()}

  body {
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
      monospace;
  }
`;

export const Wrapper = styled.div`
    margin: 0 auto;
    max-width: 1250px;
    padding: 100px 1rem;
`;

export const Link = styled(NavLink)`
    color: initial;
    text-decoration: none;
`;

export const Title = styled.h4`
    font-weight: 500;
    padding: 0 0 2rem;
`;
