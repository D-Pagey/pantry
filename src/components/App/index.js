import React, { useState, useEffect } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Media from 'react-media';
import { firebase } from '../../services';
import ProviderFirestore from '../ProviderFirestore';
import Header from '../Header';
import PageHome from '../PageHome';
import AddFoodForm from '../AddFoodForm';
import MobileNavbar from '../MobileNavbar';
import FoodTable from '../FoodTable';
import PageSignIn from '../PageSignIn';
import PageProfile from '../PageProfile';
import * as S from './styles';

const App = () => {
    const [user, setUser] = useState({ name: '', email: '', photo: '' });
    const [isAuthed, setIsAuthed] = useState(false);

    useEffect(() => {
        firebase.auth().onAuthStateChanged((user) => {
            if (user) {
                setIsAuthed(true);
                setUser({
                    name: user.providerData[0].displayName,
                    email: user.providerData[0].email,
                    photo: user.providerData[0].photoURL
                });
            } else {
                setIsAuthed(false);
            }
        });
    }, []);

    return (
        <ProviderFirestore>
            <BrowserRouter>
                <S.Wrapper>
                    <S.GlobalStyle />

                    <Header isAuthed={isAuthed} name={user.name} />

                    <Switch>
                        <Route exact path="/" component={PageHome} />
                        <Route path="/add" component={AddFoodForm} />
                        <Route
                            path="/sign-in"
                            render={(props) => (
                                <PageSignIn {...props} isAuthed={isAuthed} setUser={setUser} />
                            )}
                        />
                        <Route
                            path="/profile"
                            render={(props) => (
                                <PageProfile
                                    {...props}
                                    isAuthed={isAuthed}
                                    name={user.name}
                                    email={user.email}
                                />
                            )}
                        />
                        <Route path="/:category" component={FoodTable} />
                    </Switch>

                    <Media query="(max-width: 475px)" render={() => <MobileNavbar />} />
                </S.Wrapper>
            </BrowserRouter>
        </ProviderFirestore>
    );
};

export default App;
