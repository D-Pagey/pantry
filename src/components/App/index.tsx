import 'react-toastify/dist/ReactToastify.css';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { toast } from 'react-toastify';
import { ProviderFirebase } from '../ProviderFirebase';
import { BurgerMenu } from '../BurgerMenu';
import { Navbar } from '../Navbar';
import { Routes } from '../Routes';
import * as S from './styles';

toast.configure({
    position: 'bottom-right'
});

export const App = (): JSX.Element => (
    <ProviderFirebase>
        <BrowserRouter>
            <BurgerMenu />
            <Navbar />

            <S.Wrapper>
                <S.GlobalStyle />

                <Routes />
            </S.Wrapper>
        </BrowserRouter>
    </ProviderFirebase>
);
