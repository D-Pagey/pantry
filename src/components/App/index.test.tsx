import React from 'react';
import App from '.';

describe('App component', (): void => {
  it('should render', (): void => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
