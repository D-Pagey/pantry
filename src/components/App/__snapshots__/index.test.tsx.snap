// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render 1`] = `
.c0 {
  text-align: center;
}

<div>
  <header>
    <h1
      class="c0"
    >
      Pantry
    </h1>
  </header>
</div>
`;
